test/minimum_eigensolvers/evqe/test_evqe_operators.py

```diff
@@ -0,0 +1,208 @@
+from qiskit_optimization.translators import from_docplex_mp, to_ising
+from qiskit_optimization.converters import IntegerToBinary
+from qiskit.quantum_info import Operator
+
+from queasars.circuit_evaluation.circuit_evaluation import OperatorCircuitEvaluator
+from queasars.minimum_eigensolvers.base.evolutionary_algorithm import OperatorContext, BasePopulationEvaluationResult
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.individual import EVQEIndividual
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.population import EVQEPopulation
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.mutation import (
+    EVQELastLayerParameterSearch,
+    EVQEParameterSearch,
+    EVQETopologicalSearch,
+    EVQELayerRemoval,
+)
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.speciation import EVQESpeciation
+from queasars.minimum_eigensolvers.evqe.evolutionary_algorithm.selection import EVQESelection
+
+
+class TestEVQEOperators:
+    client: Optional[Client] = None
+    evaluator: Optional[OperatorCircuitEvaluator] = None
+    operator: Optional[Operator] = None
+
+    @pytest.fixture
+    def initial_population(self) -> EVQEPopulation:
+        return EVQEPopulation.random_population(
+            n_qubits=4, n_layers=2, n_individuals=10, randomize_parameter_values=False, random_seed=0
+        )
+
+    @pytest.fixture
+    def dask_client(self) -> Client:
+        if self.client is None:
+            cluster = LocalCluster(processes=True, n_workers=2)
+            self.client = cluster.get_client()
+        return self.client
+
+    @pytest.fixture
+    def hamiltonian(self) -> Operator:
+        if self.operator is None:
+            optimization_problem = Model()
+            x = optimization_problem.integer_var(lb=0, ub=3, name="x")
+            y = optimization_problem.integer_var(lb=0, ub=3, name="y")
+            optimization_problem.minimize(x**2 - y**2)
+
+            quadratic_program = from_docplex_mp(model=optimization_problem)
+            integer_converter = IntegerToBinary()
+            quadratic_program = integer_converter.convert(problem=quadratic_program)
+            self.operator, _ = to_ising(quad_prog=quadratic_program)
+        return self.operator
+
+    @pytest.fixture
+    def circuit_evaluator(self, hamiltonian) -> OperatorCircuitEvaluator:
+        if self.evaluator is None:
+            estimator = Estimator()
+            estimator.set_options(seed=0)
+            self.evaluator = OperatorCircuitEvaluator(qiskit_primitive=estimator, operator=hamiltonian)
+        return self.evaluator
+
+    @pytest.fixture
+    def operator_context(self, dask_client, circuit_evaluator) -> OperatorContext:
+        return OperatorContext(
+            circuit_evaluator=circuit_evaluator,
+            dask_client=dask_client,
+            result_callback=lambda x: None,
+            circuit_evaluation_count_callback=lambda x: None,
+        )
```

Comment:
This fixture shall definitely not be cached/shared as you modify the call backs in the tests later on. I am assuming this is why you didn't add a field for the operator context above?

Reply:
Correct :)